### PR TITLE
Implemented errors container

### PR DESCRIPTION
**Pull request by matcake**
Component allows to render all form errors in a single place.

### CODE REVIEW

File: src/components/ErrorsContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    constructor(props) {
+        super(props);
+        this.getErrors = this.getErrors.bind(this);
+    }
+    getErrors() {
+        const { getAllErrors } = this.context;
+        const errorsObj = getAllErrors();
+        let errors = [];
+        Object.keys(errorsObj).forEach((errorIndex) => {
+            errorsObj[errorIndex].forEach(error => errors.push(error));
```
Comment (mprzodala) [inline src/components/ErrorsContainer.jsx]: You should add support for object and array. If you use ObjectField or ListField you will get object or array of array etc.
#### Example
```js
{
    foo: [
        { 
            bar: ['some error'],
            foo: [
                bar: ['some error2']
            ]
        }
    ]
}
```

File: src/components/ErrorsContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, { Component, PropTypes } from 'react';
+import FieldConnect from './FieldConnect';
+import ErrorField from './ErrorField';
+
+class ErrorsContainer extends Component {
+    constructor(props) {
```
Comment (mprzodala) [inline src/components/ErrorsContainer.jsx]: In this case you dont need getErrors.bind and constructor.

File: tests/ErrorsContainer.test.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { ErrorsContainer } from '../src/components';
+import { ErrorField } from '../src/components/separate';
+
+describe('ErrorsContainer', () => {
```
Comment (mprzodala) [inline tests/ErrorsContainer.test.jsx:6]: I think you should add test with case when you will receive errors from ObjectField and ListField.
#### Example
```js
{
    foo: [
        { 
            bar: ['some error'],
            foo: [
                bar: ['some error2']
            ]
        }
    ]
}
```

Comment (matcake) [reply tests/ErrorsContainer.test.jsx:6]: Done

File: src/components/ErrorsContainer.jsx (hunk 2 above)
Comment (matcake) [reply src/components/ErrorsContainer.jsx]: Removed

File: src/components/ErrorsContainer.jsx (hunk 1 above)
Comment (matcake) [reply src/components/ErrorsContainer.jsx]: Done

**Comment by matcake** [comment]
@mprzodala required changes applied. Please verify

File: src/components/ErrorsContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, { Component, PropTypes } from 'react';
+import FieldConnect from './FieldConnect';
```
Comment (mprzodala) [inline src/components/ErrorsContainer.jsx]: FieldConnect no needed.

File: src/components/ErrorsContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import ErrorField from './ErrorField';
+
+const parseErrors = (data) => {
```
Comment (mprzodala) [inline src/components/ErrorsContainer.jsx]: Please work on complexity. Now if I good see we have complexity: 7. Please move some logic to seperate functions. Complexity: 4 should be max for one function.

File: src/components/Form.jsx [hunk 6]
```diff
@@ -89,6 +90,10 @@ class Form extends React.Component {
         return this.state.errors[name] || {};
     }
 
+    getAllErrors() {
+        return this.state.errors || {};
```
Comment (mprzodala) [inline src/components/Form.jsx]: Please remove default object. Object as default in state.errors is defined in constructor.

**Pull request merged**